EasyGYM/UI/Controllers/Core/LibraryViewController.swift

```diff
@@ -14,221 +14,143 @@ import FirebaseFirestore
 
 class LibraryViewController: UIViewController {
     
+    var viewModel: LibraryViewModel!
+    
     @IBOutlet weak var searchBar: UISearchBar!
     @IBOutlet weak var workoutTableView: UITableView!
-    var favoriteWorkoutIDs: Set<String> = []
-    
     var workouts: [Workout] = []           // Array to hold all workouts fetched from Firestore
     var filteredSearchWorkouts: [Workout] = []   // Array to hold filtered workouts based on search query
     var filteredWorkouts: [Workout] = []
     var isSearching = false                // Flag to track if user is currently searching
-    var isFiltering = false
+    var isFiltering = false                // Flag to track if filtering on
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        if viewModel == nil {
+            viewModel = LibraryViewModel()
+        }
         // Set delegates for table view and search bar
         workoutTableView.delegate = self
         workoutTableView.dataSource = self
         searchBar.delegate = self
-        
-        // Fetch workouts from Firestore
-        fetchWorkouts()
-        fetchFavoriteWorkouts()
+
+        viewModel.fetchWorkouts {
+            print("Data fetched, reloading table view")
+            self.workoutTableView.reloadData()
+        }
+        viewModel.fetchFavoriteWorkouts(){
```

Comment:
Since this is the LibraryViewController, is it needed to fetch the FavoriteWorkouts here?